src/api/listing/taxonomy.proto

```diff
@@ -0,0 +1,126 @@
+  // References another Category's id as this Category's parent, can be empty
+  // for top-level categories
+  string parent_category_id = 2;
+  // A user-friendly name for the category
+  string display_name = 3;
+
+  google.protobuf.Timestamp updated_at = 4;
+  google.protobuf.Timestamp created_at = 5;
+}
+
+// AttributeSpec is the definition of an attribute on a channel.
+// The spec also provides information about how to validate Attributes and how
+// to derive their value from the catalog system.
+// The link between an AttributeSpec and a Category indicates the
+// availability of this particular spec in that ProductType.
+message AttributeSpec {
+  // This is the ID that will be used when retrieving listing data or submitting
+  // error to Zentail. It should reflect whatever meaningful identifier can be
+  // used to ingest this attribute into your system. In the past we have seen it
+  // be a numeric identifier or an xpath/jsonpath string
+  string attribute_spec_id = 1;
+
+  // Human-readable name for the attribute. Used for display purposes only.
+  string display_name = 2;
+
+  // Type specifies what form the related attribute should be rendered into
+  enum Type {
+    TYPE_UNSPECIFIED        = 0;
+    TYPE_TEXT               = 1;
+    TYPE_SELECT             = 2;
+    TYPE_NUMERIC            = 3;
+    TYPE_NUMERIC_WITH_UNITS = 4;
+    TYPE_MULTI_TEXT         = 5;
+    TYPE_MULTI_SELECT       = 6;
+    TYPE_MONEY              = 7;
+  }
+  Type type = 3;
+
+  // If the type is SELECT or MULTI_SELECT, provide the valid values here
+  repeated string valid_values = 4;
+
+  // If the type is NUMERIC_WITH_UNITS, provide the valid units here
+  repeated string valid_units = 5;
+
+  // Optional, used to specify the unit of the numeric value when the type is
+  // TYPE_NUMERIC This will allow Zentail to convert values with units in
+  // Zentail to a single numeric value if your channel does not support units
+  string unit = 6;
+
+  // Level specifies whether this AttributeSpec should be rendered
+  // on the Listing-level or the Variant-level
+  enum Level {
+    LEVEL_UNSPECIFIED = 0;
+    LEVEL_LISTING     = 1;
+    LEVEL_VARIANT     = 2;
+  }
+  Level level = 7;
+
+  // Classification specifies whether this AttributeSpec relates to
+  // Product, Inventory, or Pricing data
+  enum Classification {
+    CLASSIFICATION_UNSPECIFIED = 0;
+    CLASSIFICATION_PRODUCT     = 1;
+    CLASSIFICATION_INVENTORY   = 2;
+    CLASSIFICATION_PRICING     = 3;
+    CLASSIFICATION_LOGISTICS   = 4;
+    CLASSIFICATION_IDENTIFIER  = 5;
+  }
+  Classification classification = 8;
+
+  google.protobuf.Timestamp updated_at = 9;
+  google.protobuf.Timestamp created_at = 10;
+}
+
+// ProductType are detailed classifications dictated by the channel.
+// They should be at least as specific as the Category.
+// Generally they should be more specific than the Category.
+// ProductTypes are only used in the process of deriving ListingQuality.
```

Comment:
maybe it's a little clearer to say "Product Types are used to specify which attributes are required and recommended for a given Listing"

Reply:
like it, updated